### PR TITLE
feature/set_an_avatar

### PR DESCRIPTION
**Pull request by enakecanovic**
Option to choose an avatar using a Spinner. Avatar is being saved using SharedPreferences.

### CODE REVIEW

File: app/src/main/java/at/aau/edu/wizards/MainActivity.kt [hunk 1]
```diff
@@ -20,14 +24,16 @@ private const val DISCOVER_FRAGMENT_TAG = "DISCOVER_FRAGMENT_TAG"
...
 private const val SCOREBOARD_FRAGMENT_TAG = "SCOREBOARD_FRAGMENT_TAG"
-
 private const val SHARED_PREFERENCES_KEY = "SHARED_PREFS_USERDATA"
 
 class MainActivity : AppCompatActivity() {
 
     private lateinit var binding: ActivityMainBinding
     private lateinit var mainViewModel: MainViewModel
 
+    private lateinit var imageView: ImageView
```
Comment (lechld) [inline app/src/main/java/at/aau/edu/wizards/MainActivity.kt:34]: you should just use the `binding`, that's what it is for

File: app/src/main/java/at/aau/edu/wizards/MainActivity.kt [hunk 2]
```diff
@@ -41,10 +47,37 @@ class MainActivity : AppCompatActivity() {
 
         setContentView(binding.root)
 
+        setupAvatarSelection()
         setupUi()
         handlePermissions()
     }
 
+    private fun setupAvatarSelection() {
+        imageView = findViewById(R.id.imageView)
```
Comment (lechld) [inline app/src/main/java/at/aau/edu/wizards/MainActivity.kt:56]: binding.imageView

If you like you can even give a more descriptive name

File: app/src/main/java/at/aau/edu/wizards/MainViewModel.kt [hunk 3]
```diff
@@ -15,6 +23,26 @@ class MainViewModel (private val sharedPreferences: SharedPreferences) : ViewMod
         sharedPreferences.edit().putString(SHARED_PREFERENCE_USERNAME_KEY, username).apply()
     }
 
+    fun getAvatar(): Int {
+        return sharedPreferences.getInt(SHARED_PREFERENCE_AVATAR_KEY, selectRandomAvatar())
```
Comment (lechld) [inline app/src/main/java/at/aau/edu/wizards/MainViewModel.kt:27]: always random? Maybe it makes sense to keep preference of user

File: app/src/test/java/at/aau/edu/wizards/MainViewModelTests.kt [hunk 4]
```diff
@@ -44,6 +51,35 @@ class MainViewModelTests {
         then(editor).should().putString(SHARED_PREFERENCE_USERNAME_KEY, username)
     }
 
+    // AVATAR
+
+    @Test
```
Comment (lechld) [inline app/src/test/java/at/aau/edu/wizards/MainViewModelTests.kt:56]: nice to see some tests

File: app/src/main/java/at/aau/edu/wizards/MainViewModel.kt (hunk 3 above)
Comment (enakecanovic) [reply app/src/main/java/at/aau/edu/wizards/MainViewModel.kt:27]: This function is defined so, that the random avatar will be generated only the first time when the user starts the game. SharedPrefs are used here to remember the first choice the player made, which will be shown every other time he starts the game.

**Pull request merged**